src/main/java/utilities/Id.java

```diff
@@ -0,0 +1,156 @@
+/* *********************************************************************** *
```

Comment:
The main advantages of the Id are as follows: first, it allows to use both Integer and String at the same time with "Id"; Second, it is easier to recognize the object for which an Id is used for. For instance, given two similar variables, Map<Integer, Double> averageHouseholdIncome; Map<Integer,Double> averageRegionHouseholdIncome, users may be a bit of confused about what the ID exactly is, RegionID, HouseholdID or something else. It is easy for developers to distinguish between these two variables, but users need to think about it carefully. However, if we use the class of ID here, the variables will become Map<Id<Household>, Double> averageHouseholdIncome and Map<Id<Region>,Double> averageRegionHouseholdIncome, users can easily distinguish between them without any comments: the former uses a household Id and the latter a region Id. The second advantage is significant, especially in large Java projects. However, our current project is relatively small and the benefits of using the class of Id is small. So it is fine to use the way you suggested (because there are currently not many similar variables involved), but from the long-term perspective, the class of Id might be helpful and makes the codes more readable to potential users.

Reply:
See comment about Matrix class. Just to be sure: does keeping ID class and Matrix class or removing them imply many changes in the code? As in, is adding and removing them a big hassle or is it just small localised changes? I'm still not really convinced of how useful this class and the Matrix class are for us now, but I guess I'm fine with keeping them as long as we are not necessarily committing too much and we can revise the decision in the future if we find they're not that useful.

Reply:
I will try to remove these two classes if the work does not take a long time.